### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by Jaifroid**
@mossroy This is a new security feature of GitHub. It works with JavaScript. A bot does some automatic code analysis after each commit to master. It could be useful. I thought we could try it out. We can always turn it off if it's not useful.

### CODE REVIEW

**Comment by Jaifroid** [comment]
(I will remove the cpp analysis, which obviously fails, leaving only JS analysis.)

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '44 12 * * 3'
+
+jobs:
+  analyze:
+    name: Analyse
```
Comment (mossroy) [inline .github/workflows/codeql-analysis.yml:25]: Why did you change this to "Analyse" instead of the original "Analyze"?

**Review by mossroy — APPROVED** [review]
This can be interesting

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (Jaifroid) [reply .github/workflows/codeql-analysis.yml:25]: Because I'm a linguist, and etymologically "analyze" is completely incorrect spelling! The root is Greek -lysis, as in analysis, dialysis, and is not the same root as other -ise/ize verbs in English, French, and Romance languages. OED is very specific that "analyze" is wrong even in US English (though you see it everywhere)!

This is just me bing a pedant. 😯

Comment (mossroy) [reply .github/workflows/codeql-analysis.yml:25]: I had no idea and trust you 100% on this
